### PR TITLE
Lagoon Monitoring

### PR DESCRIPTION
**Pull request by Schnitzel**
This adds Lagoon Monitoring:
1. a helmfile for a Prometheus stack monitoring Lagoon components (elasticsearch, fluentd, logging-operator, broker, etc.)
2. ServiceMonitors for Lagoon Components so it can be picked up by LagoonMonitoring

### CODE REVIEW

File: charts/lagoon-core/templates/broker.servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+metadata:
+  name: {{ include "lagoon-core.broker.fullname" . }}
+  labels:
+    {{- include "lagoon-core.broker.labels" . | nindent 4 }}
+spec:
+  endpoints:
+    - port: metrics
+  namespaceSelector:
+    matchNames:
+      - lagoon-core
```
Comment (smlx) [inline charts/lagoon-core/templates/broker.servicemonitor.yaml:13]: I think this should be something like:
```suggestion
      - {{ .Release.Namespace }}
```

File: charts/lagoon-logging/templates/logs-dispatcher.servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{{- if .Values.logsDispatcher.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "lagoon-logging.logsDispatcher.fullname" . }}
+  labels:
+    {{- include "lagoon-logging.logsDispatcher.labels" . | nindent 4 }}
+    monitoring.lagoon.sh/monitorMe: 'true'
```
Comment (smlx) [inline charts/lagoon-logging/templates/logs-dispatcher.servicemonitor.yaml:8]: Interesting, is this currently used?

File: helmfiles/lagoon-monitoring/auth.yaml.gotmpl [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  paths: ["/"]
+  hosts:
+    - {{ .Values.auth.host }}
+  tls:
+    - secretName: {{ .Values.auth.host }}-tls
+      hosts:
+        - {{ .Values.auth.host }}
+
+config:
+{{- toYaml .Values.auth.vouchConfig | nindent 2 }}
```
Comment (smlx) [inline helmfiles/lagoon-monitoring/auth.yaml.gotmpl:14]: Missing a newline here.

File: helmfiles/lagoon-monitoring/values.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+auth:
+  enabled: false
+  host: auth.monitoring.example.com
+  config:
+    vouch:
+      domains: []
+      allowAllUsers: false
+      whiteList: []
+      jwt:
+        secret: super-secret-stuff
```
Comment (smlx) [inline helmfiles/lagoon-monitoring/values.yaml:11]: So is the idea that this file is just example values that any user of the `helmfile.yaml` would need to override? If so maybe we could add a comment at the top?

**Review by smlx — CHANGES_REQUESTED** [review]
A couple of minor comments, but generally looks great! :tada:

File: charts/lagoon-logging/templates/logs-dispatcher.servicemonitor.yaml (hunk 2 above)
Comment (Schnitzel) [reply charts/lagoon-logging/templates/logs-dispatcher.servicemonitor.yaml:8]: @smlx you mean the `monitoring.lagoon.sh/monitorMe` ? Yes it's used by lagoon-monitoring prometheus
https://github.com/uselagoon/lagoon-charts/pull/133/files#diff-31c0db3ddbe675f8065e5156d8db158ca81b36a321026a4d0488df167841688aR259

File: helmfiles/lagoon-monitoring/values.yaml (hunk 4 above)
Comment (Schnitzel) [reply helmfiles/lagoon-monitoring/values.yaml:11]: ah yes, will add a comment :)

**Review by smlx — APPROVED** [review]
LGTM once CI passes

maybe we should squash + merge since there are a lot of commits here?

**Comment by Schnitzel** [comment]
> maybe we should squash + merge since there are a lot of commits here?
haha, yes definitely :)

**Pull request merged**